src/YetAnotherHttpHandler.Unity/Assets/Editor/Scripts/PackageExporter.cs

```diff
@@ -1,34 +1,67 @@
 #if UNITY_EDITOR
 
+using Needle.HybridPackages;
 using System;
+using System.Collections.Generic;
 using System.IO;
 using System.Linq;
 using UnityEditor;
-using UnityEngine;
+using UnityEditor.PackageManager.Requests;
 
 public static class PackageExporter
 {
+    private static ListRequest s_listRequest;
+
     [MenuItem("Tools/Export Unitypackage")]
     public static void Export()
     {
-        PackDependencies("Cysharp.Net.Http.YetAnotherHttpHandler.Dependencies", "YetAnotherHttpHandler.deps.txt");
-        PackDependencies("Grpc.Net.Client.Dependencies", "Grpc.Net.Client.deps.txt");
+        EditorUtility.DisplayProgressBar("Packges", "Exporting packages, please wait...", 0);
+
+        s_listRequest = UnityEditor.PackageManager.Client.List(offlineMode: false, includeIndirectDependencies: true);
+
+        EditorApplication.update += OnEditorUpdate;
+    }
+
+    private static void OnEditorUpdate()
+    {
+        if (s_listRequest == null || !s_listRequest.IsCompleted)
+        {
+            return;
+        }
+
+        EditorApplication.update -= OnEditorUpdate;
+
+        ExportPackage("com.cysharp.yetanotherhttphandler", "Cysharp.Net.Http.YetAnotherHttpHandler.Dependencies");
+        ExportPackage("org.nuget.grpc.net.client", "Grpc.Net.Client.Dependencies");
+
+        EditorUtility.ClearProgressBar();
     }
 
-    static void PackDependencies(string unityPackageName, string dependenciesListFileName)
+    private static void ExportPackage(string packageName, string unityPackageName)
     {
-        Debug.Log($"Creating package '{unityPackageName}'...");
-        var pluginsDir = Path.Combine(Application.dataPath, "Plugins");
-        var libraryNames = File.ReadAllLines(Path.Combine(pluginsDir, dependenciesListFileName));
-        Debug.Log($"Includes library: {string.Join(';', libraryNames)}");
-
-        var exportPath = $"./{unityPackageName}.unitypackage";
-        AssetDatabase.ExportPackage(
-            libraryNames.Select(x => $"Assets/Plugins/{x}").ToArray(),
-            exportPath,
-            ExportPackageOptions.Recurse);
-
-        UnityEngine.Debug.Log("Export complete: " + Path.GetFullPath(exportPath));
+        List<UnityEditor.PackageManager.PackageInfo> dependencies = new();
+
+        UnityEditor.PackageManager.PackageInfo packageInfo = s_listRequest.Result.First(p => p.name.Equals(packageName, StringComparison.Ordinal));
+
+        GetAllDependencies(packageInfo, dependencies);
+
+        string exportPath = $"./{unityPackageName}.unitypackage";
+
+        UnitypackageExporter.ExportUnitypackage(dependencies, exportPath);
```

Comment:
As far as I've tried, it seems like I can export the package with the code below even without using HybridPackage.
```csharp
AssetDatabase.ExportPackage(dependencies.Select(x => x.assetPath).ToArray(), exportPath, ExportPackageOptions.Recurse);
```
Do you have any concerns about this?

Reply:
The truth is that I had not tried it, since I thought that with the Unity API it was not possible to export packages as a *.unitypackage and that's why https://github.com/needle-tools/hybrid-packages existed.

We can try this way without that dependency, but I still have the doubt that if it exists it is for something 😅